src/Stack/Types/TemplateName.hs

```diff
@@ -106,3 +125,24 @@ templateName (TemplateName prefix _) = prefix
 -- | Get the path of the template.
 templatePath :: TemplateName -> TemplatePath
 templatePath (TemplateName _ fp) = fp
+
+defaultRepoUser :: Text
+defaultRepoUser = "commercialhaskell"
+
+-- | Parses a template path of the form @github:user/template@.
+parseRepoPath :: String -> Maybe RepoTemplatePath
+parseRepoPath s =
+  case T.splitOn ":" (T.pack s) of
+    ["github"    , rest] -> parseRepoPathWithService Github rest
+    ["gitlab"    , rest] -> parseRepoPathWithService Gitlab rest
+    ["bitbucket" , rest] -> parseRepoPathWithService Bitbucket rest
+    _                    -> Nothing
+
+-- | Parses a template path of the form @user/template@, assuming the default service.
+parseRepoPathWithService :: RepoService -> Text -> Maybe RepoTemplatePath
+parseRepoPathWithService service path =
+  case T.splitOn "/" path of
+    [user, name] -> Just $ RepoTemplatePath service user name
+    [name]       -> Just $ RepoTemplatePath service defaultRepoUser name
```

Comment:
Deleting this line and `defaultRepoUser` should change the behavior as described above regarding `commercialhaskell` on other services.

Reply:
Not quite. That would prevent `commercialhaskell` being used as the default for other services. But it would also break the simple case where only a template name is supplied. Another function will have to be created to handle that case.

Reply:
Oh right, we end up here in the case where just a single name is given, thanks.